### PR TITLE
Refactor DataDrivenNode functionality

### PR DESCRIPTION
**Pull request by Jman420**
The goal of this PR is to provide support for multiple data driven nodes in a URI such that URIs such as http://example.org/path/to/ddn/123/another/ddn/987/ and http://example.org/path/to/ddn/456/another/ddn/789/ can be replaced with http://example.org/path/to/ddn/�node1�/another/ddn/�node2�/ to treat both URIs as a single endpoint.

Changes to accomplish goal:

Separate concepts of StructuralParameters and DataDrivenNodes
Separate Context UI Panels for StructuralParameters and DataDrivenNodes
Introduce a Tree Structure for representing DataDrivenNodes
Update StandardParameterParser logic to work with DataDrivenNodes Tree Structure

Supporting Help Docs PR : https://github.com/zaproxy/zap-core-help/pull/334

### CODE REVIEW

**Comment by kingthorin** [comment]
Related: https://github.com/zaproxy/zaproxy/issues/5616

**Comment by Jman420** [comment]
@zaproxy/team-zaproxy Any comments or suggestions on these changes?

I'm not sure how to fix the japicmp build errors... I did indeed remove a bunch of code files due to my refactoring, so I would expect there to be differences between the two builds.  Any guidance there would be helpful.

Thanks!

**Comment by thc202** [comment]
The breaking changes need to be suppressed in `zap/zap.gradle.kts`, under the `japicmp` task. Run the task to generate the report and know what needs to be suppressed. (Better rebase first on the `main` branch, the file was changed.)

I'd say that this is breaking too much though, add-ons might be using some of the methods/classes being removed. Ideally the old classes/methods should be deprecated first and removed only after some releases.

Some loose comments:
 - Needs tests;
 - Commit History should be tidied up, hard to follow.

**Comment by Jman420** [comment]
@thc202 Ok, I can bring the classes I removed back... I assume that would include the UI classes I removed too, although that will add some complexity to the UI since there will be two places that Data Driven Nodes & Structural Parameters are handled.

How does the team go about marking classes as deprecated?  Is there an attribute to use?

I'll take a stab at cleaning up the commit history... I'm not very familiar with rewriting commit histories.

**Comment by thc202** [comment]
They don't need to be used just don't remove them, you can add the annotation `@Deprecated` and the JavaDoc tag `@deprecated` (e.g. `HttpMessageList` class).

**Comment by Jman420** [comment]
> They don't need to be used just don't remove them, you can add the annotation `@Deprecated` and the JavaDoc tag `@deprecated` (e.g. `HttpMessageList` class).

Got ya.  Thanks!  Took a bit of a break from this over the holidays.  Plan on diving back in again next week.

**Comment by Jman420** [comment]
Moving PR to main branch and cleaning up commit history

**Comment by thc202** [comment]
That can be done in this pull request.
